### PR TITLE
Recover from panics in cron jobs

### PR DESCRIPTION
**Pull request by blackskad**
When a cronjob panics, it takes the whole program down with it. This patch recovers after the panic and prints a message containing the data included in the panic.

### CODE REVIEW

**Comment by elgs** [comment]
A library should not print anything. It should return the error instead. 

Also I didn't see anywhere in the library panics. It should not be the library's responsibility to swallow the panic if the job panics. It should be the job's responsibility to handle the panic.

**Comment by blackskad** [comment]
The net/http library has similar behaviour:
- It recovers the panic, assuming the effects are limited to that request
- A stack trace is printed to the server log

https://golang.org/pkg/net/http/#Serve

**Comment by robfig** [comment]
The comparison with net/http for use of recover and logging seems reasonable. Happy to merge this if you change it to use a package-level logger & a format more like what's in net/http:
https://golang.org/src/net/http/server.go?s=58455:58504#L1382

e.g. 
```go
logger.Println("cron: panic running job:%v\n%v", err, stacktrace)
```

**Comment by blackskad** [comment]
Thank you for the feedback. I added a cron-level logger as suggested.

File: cron.go [hunk 1]
```diff
@@ -127,6 +131,18 @@ func (c *Cron) Start() {
 	go c.run()
 }
 
+func (c *Cron) runWithRecovery(j Job) {
+	defer func() {
+		if r := recover(); r != nil {
+			const size = 64 << 10
+			buf := make([]byte, size)
+			buf = buf[:runtime.Stack(buf, false)]
+			c.logf("cron: panic running job:%v\n%s", r, buf)
```
Comment (robfig) [inline cron.go]: pardon the nitpick - please add a space after the colon "job: "

**Comment by robfig** [comment]
Thank you

**Comment by blackskad** [comment]
Thank you for merging it in :)

**Pull request merged**